src/components/tx/TxAlerts.js

```diff
@@ -0,0 +1,32 @@
+import React, { useState, useEffect } from 'react';
+import { txTokenBannedMessage } from '../../messages';
+
+
+const TxAlerts = (props) => {
+  const [transaction, setTransaction] = useState(props.transaction);
+
+  useEffect(() => {
+    setTransaction(props.transaction);
+  }, [props.transaction]);
+
+  const bannedAlert = () => {
+    if (!transaction.tokens.find(token => token.meta && token.meta.banned)) {
```

Comment:
On the `TxData` component you are saving the tokens metadata in the state. Can you get token meta like this?

Or when you do 

```
return metadataApi.getDag(token.uid).then((data) => {
      token.meta = data;
      return token;
```

You are actually mutating the props object?

Reply:
Changed to a solution proposed by @andreabadesso

Reply:
That's good but this created a bug in this screen, right? You won't have `token.meta` in the `props.transaction.tokens` array. Or am I missing something?

Reply:
Ok after your last commit